protos/flyteidl/admin/common.proto

```diff
@@ -7,18 +7,26 @@ import "flyteidl/core/execution.proto";
 import "flyteidl/core/identifier.proto";
 // import "validate/validate.proto";
 
+message NamedEntityMetadata {
+    // Common description across all versions of the entity
+    // +optional
+    string description = 1;
+}
+
 // Encapsulation of fields that identifies a Flyte resource.
 // A resource can internally have multiple versions.
 message NamedEntityIdentifier {
     // Name of the project the resource belongs to.
-    string project    = 1;  // [(validate.rules).string.min_bytes = 1];
+    string project               = 1;  // [(validate.rules).string.min_bytes = 1];
     // Name of the domain the resource belongs to.
     // A domain can be considered as a subset within a specific project.
-    string domain     = 2;  // [(validate.rules).string.min_bytes = 1];
+    string domain                = 2;  // [(validate.rules).string.min_bytes = 1];
     // User provided value for the resource.
     // The combination of project + domain + name uniquely identifies the resource.
     // +optional - in certain contexts - like 'List API', 'Launch plans'
-    string name       = 3;
+    string name                  = 3;
+
+    NamedEntityMetadata metadata = 4;
```

Comment:
why is the metadata part of the identifier?  can you not create a separate message encapsulates the identifier and the metadata?

Reply:
The intention is that you should be able to list the identifiers and also have a description of what each identifier is.

Reply:
I moved the metadata out and used a message composing id and metadata, and made that the preferred type when returning a list of identifiers. ptal!